### ISSUE TITLE
[Mod Add] Weapon Master - Modular Scaling Weapons

### ISSUE THREAD

**Issue by RumekFuria** [JPAchilles/20stm#2]
https://modrinth.com/mod/weapon-master

This mod could be our answer to "Tetra Meh, Tinker isn't on Modrinth"
Offers a lot of customization option while technically just being 4 weapons

### Note
the mod has a known incompatibility with Better Combat, as listed on the modpage, but it also provides a solution
the entry is as follows:

Hit detection breaks and makes the weapons unusable due to mixin issues. Oops! You can add "weaponmaster" to the blacklist [of Better Combat] to make our weapons function normally while letting these mods edit everything else. There will not be cross-compatability on Weapon Master weapons to work in them as it would be wildly unbalanced due to how the weapons are scaled.

**Comment by Existentialism-In-A-Cup** [comment]
This mod has four weapons that cannot be enchanted but can level up aspects of itself (Damage, Attack Speed, Durability, and Looting respectively) by dealing damage. If the weapon breaks, you can repair it for free (literally, all you need to restore durability is to play a minigame. no additional material cost). The weapon also comes with a right click active ability you can choose from a pool of dozens of abilities and a passive ability that only applies when the weapon is on your main hand (not offhand). These weapons outscales every other weapon we have in the modpack at endgame, but are comparable to normal weapons early game. A fully-levelled broadsword (16 damage, 3 atk speed, 2000 durability) will outdps a sharpness V draco arcanus sword (15 damage, 1.8 atk speed, 2661 durability).

All customisation is vanity-only and doesnt affect performance but some actually make the repair minigame harder which i found amusing.

**Comment by RumekFuria** [comment]
does it take long/much to get a weapon into the endgame state?

**Comment by RumekFuria** [comment]
also... there should be abilities? It sure as hell mentioned abilities on the modpage

**Comment by RumekFuria** [comment]
To answer my own question:
- Weapon leveling is done by just killing. no shortcuts. It seems there are also increased costs to level/diminishing returns on buffs according to the small guide book
- Abilities are directly obtained from leveling, both active and passive, with a weapon being able to hold one each
there include but are not limited to:
>Heavenly Potemkin Buster
>Nefarious Elixir
>Backdash/Dash/Leap
>Matrix projectile stop and redirect
>Summon blades
>Parry

as for the blacklist... not sure
I though it was the "fallback_compatibility.json" file, as I saw nothing in the server.json5, but the entry I inserted there disappeared, so it might need to be a mini-datapack. will ask around

**Comment by RumekFuria** [comment]
oh and yeah nothing crashes when loaded with the modpack, it's fine in that regard

**Comment by RumekFuria** [comment]
I have only now realized Exis reply did mention abilities, just... briefly so I missed it

**Comment by Existentialism-In-A-Cup** [comment]
right. exp is also only earned based on damage dealt, not kills. At least, it seems that way from the testing. Killing 50 half heart ravagers isn't even close to comparing to killing just one full health (100 hp) ravager.
